### PR TITLE
[develop2] proof of concept of adding overrides to lockfiles

### PR DESCRIPTION
**Pull request by memsharded**
Via ``conan lock add`` new command

Close https://github.com/conan-io/conan/pull/10046
Close https://github.com/conan-io/conan/issues/9912

### CODE REVIEW

File: conans/cli/commands/lock.py [hunk 1]
```diff
@@ -61,3 +65,27 @@ def lock_merge(conan_api, parser, subparser, *args):
...
     result.save(lockfile_out)
     ConanOutput().info("Generated lockfile: %s" % lockfile_out)
+
+
+@conan_subcommand()
+def lock_add(conan_api, parser, subparser, *args):
+    """
+    Merge 2 or more lockfiles
+    """
+    subparser.add_argument('--requires', action="append", help='Path to lockfile to be merged')
```
Comment (lasote) [inline conans/cli/commands/lock.py]: Fix the help. What is this argument requiring exactly? has to be "name/version" or could it be any valid reference? with revision too? Would it make sense a package reference too?

**Review by lasote — COMMENTED** [review]
Looking really good.

**Comment by memsharded** [comment]
Merged develop2 with the new refactored Lockfile class.
Added the cases for python-requires and build-requires, and tests for all.

Please re-review.

**Review by czoido — APPROVED** [review]

**Review by lasote — APPROVED** [review]

**Comment by memsharded** [comment]
Seems this direction is positive. Lets move forward this for alpha.6 and remove the --require-override cli args?

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**